### PR TITLE
chore: Update docs

### PR DESCRIPTION
**Pull request by gaocegege**
Signed-off-by: Ce Gao <gaoce@caicloud.io>

**What this PR does / why we need it**:

Add links for all docs

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/1024)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,29 +10,30 @@
...
+      * [Suggestion](#suggestion)
+      * [Trial](#trial)
+      * [Worker Job](#worker-job)
+  * [Components in Katib](#components-in-katib)
+  * [Web UI](#web-ui)
+  * [API documentation](#api-documentation)
+  * [Installation](#installation)
+      * [TF operator](#tf-operator)
+      * [Pytorch operator](#pytorch-operator)
+      * [Katib](#katib-1)
```
Comment (andreyvelich) [inline README.md]: Should we name it `[Katib](#katib)` ?
The link (#katib-1) doesn't work for me.

Comment (gaocegege) [reply README.md]: Oh yes, it is a typo.

**Pull request merged**